src/System.Device.Gpio/System/Device/Gpio/Drivers/LibGpiodDriver.Linux.cs

```diff
@@ -157,14 +189,25 @@ protected internal override void SetPinMode(int pinNumber, PinMode mode)
             int requestResult = -1;
             if (_pinNumberToSafeLineHandle.TryGetValue(pinNumber, out SafeLineHandle pinHandle))
             {
+                int flags;
                 string consumer = pinNumber.ToString();
-                if (mode == PinMode.Input)
-                {
-                    requestResult = Interop.libgpiod.gpiod_line_request_input(pinHandle, consumer);
-                }
-                else
+
+                switch (mode)
                 {
-                    requestResult = Interop.libgpiod.gpiod_line_request_output(pinHandle, consumer);
+                    case PinMode.Input:
+                        requestResult = Interop.libgpiod.gpiod_line_request_input(pinHandle, consumer);
+                        break;
+                    case PinMode.InputPullDown:
+                        flags = (int) RequestFlag.GPIOD_LINE_REQUEST_FLAG_BIAS_PULL_DOWN;
+                        requestResult = Interop.libgpiod.gpiod_line_request_input_flags(pinHandle, consumer, flags);
```

Comment:
Same comment as above. I assume this requires 1.5 for libgpiod, so in that case you probably want to add this in a try catch block as it is likely that the method won't bind and throw an exception here.

Reply:
Yes ok, my doubt is if we check this early on the IsPinModeSupported

Reply:
If we add this check in IsPinModeSupported then we theoretically won't need to add it again here (although wrapping the native call might still be helpful for other reasons) since when you call on GpioController.SetPinMode, it will first callinto IsPinModeSupported, and will call this driver method only if that previous call returned true.

Reply:
Thanks for review. I'm making all the versions checks on the IsPinModeSupported.